File: src/main/java/com/auction/exception/ErrorCode.java
package com.auction.exception;
//ĐỂ ĐỊNH NGHĨA NHỮNG LỖI GẶP TRONG HỆ THỐNG
public enum ErrorCode {

    // Auction
    AUCTION_NOT_FOUND,
    AUCTION_INVALID_STATE,
    AUCTION_ALREADY_ENDED,

    // Bid
    BID_TOO_LOW,
    INVALID_BID,

    // Item
    INVALID_ITEM,
    ITEM_DUPLICATE,
    ITEM_NOT_FOUND,

    // User
    USER_NOT_FOUND,
    UNAUTHORIZED,

    // System
    CONCURRENCY,
    INVALID_INPUT
}

File: common/src/main/java/com/auction/common/exception/ErrorCode.kt
package com.auction.common.exception;

// thiết lập các mã lỗi
public enum ErrorCode {

    // auction
    AUCTION_NOT_FOUND,
    AUCTION_INVALID_STATE,
    AUCTION_ALREADY_ENDED,

    // bidder
    BID_TOO_LOW,
    INVALID_BID,

    // item
    INVALID_ITEM,
    ITEM_DUPLICATE,
    ITEM_NOT_FOUND,

    // user
    USER_NOT_FOUND,
    UNAUTHORIZED,

    // system
    CONCURRENCY,
    INVALID_INPUT,
    INTERNAL_ERROR, // Thêm dấu phẩy ở đây

    // transaction (Thêm mới cho nạp/rút)

    TRANSACTION_FAILED,
    INSUFFICIENT_BALANCE
}